drivers/ng_at86rf2xx/ng_at86rf2xx_netdev.c

```diff
@@ -0,0 +1,597 @@
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_ng_at86rf2xx
+ * @{
+ *
+ * @file
+ * @brief       Netdev adaption for the AT86RF2xx drivers
+ *
+ * @author      Thomas Eichinger <thomas.eichinger@fu-berlin.de>
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include "net/ng_ieee802154.h"
+#include "net/ng_netbase.h"
+#include "ng_at86rf2xx.h"
+#include "ng_at86rf2xx_netdev.h"
+#include "ng_at86rf2xx_internal.h"
+#include "ng_at86rf2xx_registers.h"
+
+#define ENABLE_DEBUG (1)
+#include "debug.h"
+
+
+/* TODO: generalize and move to ng_ieee802154 */
+static size_t _make_data_frame_hdr(ng_at86rf2xx_t *dev, uint8_t *buf,
+                                   ng_netif_hdr_t *hdr)
+{
+    int pos = 0;
+
+    /* we are building a data frame here */
+    buf[0] = NG_IEEE802154_FCF_TYPE_DATA;
+    /* default to wants_ack TODO: provide means to specify */
+    buf[0] |= NG_IEEE802154_FCF_ACK_REQ;
+    /* default to compress pan TODO: provide means to specify src PAN*/
+    buf[0] |= NG_IEEE802154_FCF_PAN_COMP;
+    if (hdr->dst_l2addr_len == 2) {
+        /* default to 2 byte addresses for src and dst */
+        buf[1] |= 0x88;
+        pos = 3;
+        buf[pos++] = (uint8_t)((dev->pan) & 0xff);
+        buf[pos++] = (uint8_t)((dev->pan) >> 8);
+        /* set destination address */
+        uint8_t *dst_addr = ng_netif_hdr_get_dst_addr(hdr);
+        buf[pos++] = dst_addr[1];
+        buf[pos++] = dst_addr[0];
+        /* set source address */
+        buf[pos++] = (uint8_t)((dev->addr_short) >> 8);
+        buf[pos++] = (uint8_t)((dev->addr_short) & 0xff);
+    }
+    else if (hdr->dst_l2addr_len == 8) {
+        /* default to use long address mode for src and dst */
+        buf[1] = 0xcc;
+        /* set destination address located directly after ng_ifhrd_t in memory */
+        pos = 3;
+        buf[pos++] = *((uint8_t *)(hdr + sizeof(ng_netif_hdr_t) + 8));
+        buf[pos++] = *((uint8_t *)(hdr + sizeof(ng_netif_hdr_t) + 7));
+        buf[pos++] = *((uint8_t *)(hdr + sizeof(ng_netif_hdr_t) + 6));
+        buf[pos++] = *((uint8_t *)(hdr + sizeof(ng_netif_hdr_t) + 5));
+        buf[pos++] = *((uint8_t *)(hdr + sizeof(ng_netif_hdr_t) + 4));
+        buf[pos++] = *((uint8_t *)(hdr + sizeof(ng_netif_hdr_t) + 3));
+        buf[pos++] = *((uint8_t *)(hdr + sizeof(ng_netif_hdr_t) + 2));
+        buf[pos++] = *((uint8_t *)(hdr + sizeof(ng_netif_hdr_t) + 1));
+        /* set source address */
+        buf[pos++] = (uint8_t)((dev->addr_long) >> 56);
+        buf[pos++] = (uint8_t)((dev->addr_long) >> 48);
+        buf[pos++] = (uint8_t)((dev->addr_long) >> 40);
+        buf[pos++] = (uint8_t)((dev->addr_long) >> 32);
+        buf[pos++] = (uint8_t)((dev->addr_long) >> 24);
+        buf[pos++] = (uint8_t)((dev->addr_long) >> 16);
+        buf[pos++] = (uint8_t)((dev->addr_long) >> 8);
+        buf[pos++] = (uint8_t)((dev->addr_long) & 0xff);
+    }
+    /* unsupported address length */
```

Comment:
IETF documents also only speak about 0xffff as broadcast address for IEEE 802.15.4.